### ISSUE TITLE
Simulation of multiple NGSIM environments in parallel fails

### ISSUE THREAD

**Issue by CyberSY** [huawei-noah/SMARTS#1107]
## **BUG REPORT**
**High Level Description**
The environments fail when creating multiple parallel NGSIM environments by multiprocessing.The errors occurred in reset and step functions.It works fine when creating multiple environments in a single process.

**SMARTS version**
0.4.17

**Error logs and screenshots**
reset function error:
![49a4f6511ef4276954bf79cfa08c023](https://user-images.githubusercontent.com/52485683/137121655-4d8acf49-b116-4dd2-b193-8c2992f52085.png)
step function error:
![98c99baa045ba94e07de330fb73af51](https://user-images.githubusercontent.com/52485683/137122014-ea9bc0be-439b-4595-a3f6-d0ff1cd73546.png)

**System information**

- Operating system:Ubuntu 16.04.5
- Python version :3.7

**Comment by CyberSY** [comment]
> > From the traceback, the error appears to originate from the multiprocessing communication pipe.
> > Instead of writing custom parallel environment code, do consider using our `ParallelEnv()` wrapper from #1087 to obtain rollouts from multiple SMARTS based environments in parallel.
> > Could you provide a
> > 
> > 1. minimal working example of the code used to run multiple parallel ngsim environments?
> > 2. minimal working example of the code used to run a single ngsim environment?
> 
> Hi,@Adaickalavan. We are still trying to reproduce the bug using a minimal code.We found that it might be a training-specific problem because the minimal code works fine. Here is the code we used to create NGSIM environments: https://github.com/zbzhu99/SMARTS_Imitation/blob/master/smarts-imitation/smarts_imitation/envs/traffic_gym.py


Hi,@Adaickalavan.We now successfully reproduced the bug.The error occurred when we create a pure environment before we create the multiple parallel ngsim environments.Here is the code:
https://github.com/CyberSY/SMARTS_Imitation/blob/bug_test/bug_reproduce_zbzhu.py

**Comment by Adaickalavan** [comment]
@CyberSY 
1. Please provide the steps to install dependencies and to run the "bug_reproduce_zbzhu.py" code.
1. Please provide the error thrown by this code when running it.

**Comment by CyberSY** [comment]
@Adaickalavan ：
1.We have removed the dependency of smarts_imitation and now it can be run in a pure SMARTS environment.
2.Error logs:
<img width="1266" alt="0ea3cbadf2f50e25c5e9a9b1e7ec4f1" src="https://user-images.githubusercontent.com/52485683/143583049-d114a102-bcc0-4706-8c93-9b629142109e.png">

3.You can use https://github.com/CyberSY/SMARTS_Imitation/tree/bug_test/smarts-imitation/ngsim for the scenario

**Comment by Adaickalavan** [comment]
I believe the problem is resolved and thus this issue is being closed. Please feel free to reopen this issue if your problem persists or open a new issue for other questions.
